From f21bc2bd68ccbb537c4a2f197783a0af2c1cdff2 Mon Sep 17 00:00:00 2001
From: Thu Nguyen <thu@os.amperecomputing.com>
Date: Wed, 1 Nov 2023 10:17:57 +0700
Subject: [PATCH] Update NetworkID D-Bus property type

Commit
`https://github.com/openbmc/phosphor-dbus-interfaces/commit/5e7bca41c4460abb6f513ae8e0a67c551d6192a5`
changed the type of NetworkID D-Bus property to unsigned int. This
commit updates type of NetworkID in mctp code.

Signed-off-by: Thu Nguyen <thu@os.amperecomputing.com>
---
 src/mctpd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mctpd.c b/src/mctpd.c
index b89b01b..88a8506 100644
--- a/src/mctpd.c
+++ b/src/mctpd.c
@@ -2328,7 +2328,7 @@ static int bus_endpoint_get_prop(sd_bus *bus,
 	int rc;
 
 	if (strcmp(property, "NetworkId") == 0) {
-		rc = sd_bus_message_append(reply, "i", peer->net);
+		rc = sd_bus_message_append(reply, "u", peer->net);
 	} else if (strcmp(property, "EID") == 0) {
 		rc = sd_bus_message_append(reply, "y", peer->eid);
 	} else if (strcmp(property, "SupportedMessageTypes") == 0) {
@@ -2348,7 +2348,7 @@ static int bus_endpoint_get_prop(sd_bus *bus,
 static const sd_bus_vtable bus_endpoint_vtable[] = {
 	SD_BUS_VTABLE_START(0),
 	SD_BUS_PROPERTY("NetworkId",
-			"i",
+			"u",
 			bus_endpoint_get_prop,
 			0,
 			SD_BUS_VTABLE_PROPERTY_CONST),
-- 
2.17.1

